test/integration/short/prepare-simulator-tests.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+var assert = require('assert');
+var util = require('util');
+
+var Client = require('../../../lib/client');
+var types = require('../../../lib/types/index');
+var utils = require('../../../lib/utils');
+var helper = require('../../test-helper');
+var reconnection = require('../../../lib/policies/reconnection');
+var simulacronHelper = require('../simulacron-helper');
+
+describe('Client #prepare', function () {
+  this.timeout(20000);
+  describe('Preparing statements on nodes behavior', function () {
+    var nodes = "5";
+    var clusterName = "simulacronTest";
+    var cassandraVersion = "3.10";
+    var sCluster = null;
+    var client = null;
+    var query = util.format('SELECT * FROM ks.table1 WHERE id1 = ?');
+    before(function (done) {
+      simulacronHelper.start(done);
+    });
+    beforeEach(function (done) {
+      utils.series(
+        [
+          function startCluster(next) {
+            sCluster = new simulacronHelper.SimulacronCluster();
+            sCluster.start(nodes, cassandraVersion, "", clusterName, true, 1, next);
+          },
+          function connectCluster(next) {
+            client = new Client({
+              contactPoints: [sCluster.dcs[0].nodes[0].address],
+              policies: {
+                reconnection: new reconnection.ConstantReconnectionPolicy(100)
+              }
+            });
+            client.connect.bind(client);
+            next();
+          },
+          function primeQuery(next) {
+            sCluster.primeQueryWithEmptyResult(query, next);
+          },
+          function clearLog(next) {
+            sCluster.clearLog(next);
+          }
+        ], done);
+    });
+    afterEach(function (done) {
+      client.shutdown.bind(client);
+      sCluster.destroy(done);
+    });
+    after(function (done) {
+      simulacronHelper.stop(done);
+    });
+    it('should prepare query on all hosts', function (done) {
+      var idRandom = types.Uuid.random();
+      client.execute(query, [idRandom], {prepare: 1}, function (err, result) {
+        assert.ifError(err);
+        assert.strictEqual(client.hosts.length, 5);
+        assert.notEqual(result, null);
+        assert.notEqual(result.rows, null);
+        utils.eachSeries(client.hosts.values(), function(host, next) {
+          sCluster.queryNodeLog(host.address, function(err, log) {
+            assert.ifError(err);
+            var prepareQuery = log.find(function(queryLog) {
+              return queryLog.type.startsWith("PREPARE");
```

Comment:
can be equals instead of startsWith right?   Would also be good to validate the query text as well.